### PR TITLE
feat: Add autoseal consensus and downloaders

### PR DESCRIPTION
**Pull request by mattsse**
This adds an automatic mining mode that mines blocks at a specified interval (uses 1s same as default clique periiod.)

The main motivation behind this is for testing (hive).

the mode is enabled via `--auto-mine`. 
This implementation consists of 3 parts:
* Consesus impl: effectively a noop
* Downloader impl: provides the pipeline (headers,bodies stages) with new headers + bodies
* Mining task: after each block period (interval): it drains the pool, creates a new block(header+body) with the transaction and sends a new ForkChoice update which triggers a new pipeline loop

It disables merkle stages because they're not required for hive tests and would make mining task more complicated

supersedes #1873

~~now blocked by #1662 #1845 - need to figure out how to announce new hashes now~~

~~blocked by #1996 #1992~~

### CODE REVIEW

**Comment by gakonst** [comment]
What exactly do we need to implement in the Beacon / sync controller PR to unblock this? @rkrasiuk

**Comment by gakonst** [comment]
Blocking PRs now merged

**Comment by gakonst** [comment]
Are we blocked on this, or is it RFR/merge?

**Comment by mattsse** [comment]
this is ready, this makes the node command a bit more verbose though

**Comment by gakonst** [comment]
Cool. @Rjected pls review, and I'll do a pass by EOD & we merge

**Review by Rjected — APPROVED** [review]
looks really good, let's ship it and stress test in hive

File: crates/consensus/auto-seal/src/task.rs
Comment (gakonst) [inline crates/consensus/auto-seal/src/task.rs:115]: move to config in followup if it ends up being an issue

**Review by gakonst — APPROVED** [review]

**Pull request merged**
